cuml/rf_demo.ipynb

```diff
@@ -0,0 +1,439 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Random Forest Classification\n",
+    "The Random Forest algorithm is a classification algorithm which builds several decision trees, and aggregates each of their outputs to make a prediction. This makes it more robust to overfitting. The RandomForestClassifier model implemented in the cuML library can accept the following parameters : \n",
+    "1. n_estimators: (default = 10) number of trees in the forest.\n",
+    "2. handle: cuml.Handle. If it is None, a new handle is created for this instance.\n",
+    "3. split_algo: 0 for HIST, 1 for GLOBAL_QUANTILE and 3 for SPLIT_ALGO_END (default = 0) the algorithm to determine how nodes are split in the tree.\n",
+    "4. bootstrap: (default = True) Control bootstrapping. If set, each tree in the forest is built on a bootstrapped sample with replacement. If false, sampling without replacement is done.\n",
+    "5. bootstrap_features: (default = False) Control bootstrapping for features. If features are drawn with or without replacement\n",
+    "6. rows_sample: (default = 1.0) Ratio of dataset rows used while fitting each tree.\n",
+    "7. max_depth: (default = -1) Maximum tree depth. Unlimited (i.e, until leaves are pure), if -1.\n",
+    "8. max_leaves: (default = -1) Maximum leaf nodes per tree. Soft constraint.  Unlimited, if -1.\n",
+    "9. max_features: (default = 1.0) Ratio of number of features (columns) to consider per node split.\n",
+    "10. n_bins: (default = 8) Number of bins used by the split algorithm.\n",
+    "11. min_rows_per_node: (default = 2) The minimum number of samples (rows) needed to split a node.\n",
+    "\n",
+    "The methods that can be used with the RandomForestClassifier are:\n",
+    "1. fit: Fit the model with X and y.\n",
+    "1. get_params: Sklearn style return parameter state\n",
+    "1. predict: Predicts the y for X.\n",
+    "1. set_params: Sklearn style set parameter state to dictionary of params.\n",
+    "\n",
+    "\n",
+    "The model accepts only numpy arrays or cudf dataframes as the input. In order to convert your dataset to cudf format please read the cudf documentation on https://rapidsai.github.io/projects/cudf/en/latest/. For additional information on the RandomForest model please refer to the documentation on https://rapidsai.github.io/projects/cuml/en/latest/index.html\n",
+    "\n",
+    "This notebook demonstratrates fitting a RandomForestClassifier on the Higgs data and comparing the performance (accuracy and speed) with sklearn's parallel RandomForestClassifier implementation."
```

Comment:
Should probably explain what the Higgs data is and where it comes from very briefly - like one sentence.

Reply:
After giving general information about the lab (which you did), usually explaining the code right before each cell makes the notebook more readable. For instance, these parameter details can be moved to right before you create the RandomForestClassifier.